### DISCUSSION TITLE
How to check if the user switched to Administrative Access mode?

### DISCUSSION THREAD

**Discussion by RamonGiovane** [cockpit-project/cockpit#21153] [Q&A] [answered]
I created a plugin for Cockpit which requires root user or at least `sudo` access. I'm struggling to understand how could I display an info message requiring a non root user to enter the Administrative Access mode.

The developers guide mentions the `cockpit.perrmission` API and I gave it a try. No success. It wasn't clear for me how to use it.

![image](https://github.com/user-attachments/assets/32493df8-c558-41bc-9217-42a72f996221)

**Comment by martinpitt** [comment, marked as answer]
I.e. you want your page to show an alert or similar when not in privileged mode? This is quite common indeed. We have a little [superuser.js](https://github.com/cockpit-project/cockpit/blob/main/pkg/lib/superuser.js) library (which you can pull in from cockpit like starter-kit does, or just copy). This has a `superuser.allowed` boolean(ish) property, and also a callback when the superuser state changes, so that you page can react.

cockpit.permission is older code, and TBH I'm not sure right now how superuser.js is "better" -- but this should also work:
```
const permission = cockpit.permission({ admin: true });
const amISuperuser = permission.allowed;
permission.addEventListener("changed", () => console.log("it changed!"));
```

**Reply by RamonGiovane** [reply]
That's what I needed. Thanks for the quick response.

I ended up using the `cockpit.permission.` I found it safer since it's in the documentation while I couldn't find superuser there.

I just want to point out that the docs did not state which options we can use on `permission`, so your code snippet was indispensable for me.

**Reply by martinpitt** [reply]
True that! Fixed in https://github.com/cockpit-project/cockpit/pull/21197
